Puppet 4 compliant with rspec tests

OandOSpaceboy opened the pull request.

Commit:
porten naar puppet4 en rspec testen #SYSADMIN-252

Signed-off-by: Frank Maes <franmaes.Maes@UGent.be>

Commit:
typo in j-policy.z-badmx.txt

Signed-off-by: Frank Maes <franmaes.Maes@UGent.be>

Commit:
restart the service jchmail before cdb-update

Signed-off-by: Frank Maes <franmaes.Maes@UGent.be>

Commit:
changed rspec test for cdbupdate.pp

Signed-off-by: Frank Maes <franmaes.Maes@UGent.be>

Commit:
reload the service jchmail before cdbupdate

Signed-off-by: Frank Maes <franmaes.Maes@UGent.be>

Commit:
include changes in cdbupdate.pp

Signed-off-by: Frank Maes <franmaes.Maes@UGent.be>

Commit:
some cleanup

Signed-off-by: Frank Maes <franmaes.Maes@UGent.be>

Commit:
added debian-82-x64 UGent acceptance tests

Signed-off-by: Frank Maes <franmaes.Maes@UGent.be>

rgevaert commented:
Frank, vergeet je de README file ook niet te updaten aub

Commit:
added params and templates to overide j-chkmail config

Signed-off-by: Frank Maes <franmaes.Maes@UGent.be>

Commit:
only use the hash 'config' to override the default settings

Signed-off-by: Frank Maes <franmaes.Maes@UGent.be>

rgevaert commented:
Just what I wanted to say :+1: 
- Doe je ook nog parameter validatie van String, Hash, ... 
- Acceptance tests voor opties die overriden kunnen worden 'j-ndc DUMPCF | grep ...'
- testen voor de defined resources jchkmail::cdb en jchkmail::jconf
  Thanks

Commit:
Added Puppet version 4.5.0 for acceptance tests
Added Beaker Puppet Agent Version 1.5.1
Changed Puppet Lint config
Added data type declaration

Signed-off-by: Frank Maes <franmaes.Maes@UGent.be>

Commit:
added rspec tests for defined classes

Signed-off-by: Frank Maes <franmaes.Maes@UGent.be>

Commit:
acceptance test j-ndc config

Signed-off-by: Frank Maes <franmaes.Maes@UGent.be>

OandOSpaceboy commented:
Extra rspec en acceptance testen.
